libr/bin/dwarf.c

```diff
@@ -190,6 +190,72 @@ static const char *dwarf_langs[] = {
 	[DW_LANG_Fortran08] = "Fortran08"
 };
 
+typedef struct {
+	const ut8 *buf;
+	st64 len;
+	bool big_endian;
+	size_t address_size;
+} data_xtor;
+
+void data_xtor_init(data_xtor *e, const ut8 *buf, st64 len, bool big_endian) {
+	e->buf = buf;
+	e->len = len;
+	e->big_endian = big_endian;
+}
+
+ut8 data_xtor_u8(data_xtor *e, st64 *off) {
+	ut8 ret = 0;
+	if (*off < e->len) {
+		ret = e->buf[*off];
+		*off += 1;
+	}
+	return ret;
+}
+
+ut16 data_xtor_u16(data_xtor *e, st64 *off) {
+	ut16 ret = 0;
+	if (*off + 2 <= e->len) {
+		ret = *(uut16*)(e->buf + *off);
+		if (R_SYS_ENDIAN != (int)e->big_endian)
+			ret = r_swap_ut16 (ret);
```

Comment:
same here, we have apis to read from a buffer into a big/little number of different sizes